pkg/operator/controller/ingress/status.go

```diff
@@ -457,6 +459,55 @@ func computeDeploymentReplicasAllAvailableCondition(deployment *appsv1.Deploymen
 	}
 }
 
+// computeDeploymentRollingOutCondition computes the ingress controller's
+// "DeploymentRollingOut" status condition by examining the number of updated
+// replicas reported in the deployment's status. The "DeploymentRollingOut"
+// condition is true if the number of updated replicas is not equal to the number
+// of expected or available replicas.
+// See Reference: https://github.com/kubernetes/kubectl/blob/master/pkg/polymorphichelpers/rollout_status.go
+func computeDeploymentRollingOutCondition(deployment *appsv1.Deployment) operatorv1.OperatorCondition {
+	// If have replicas is less than want replicas, then we are waiting for replicas to be updated.
+	if deployment.Spec.Replicas != nil && deployment.Status.UpdatedReplicas < *deployment.Spec.Replicas {
+		return operatorv1.OperatorCondition{
+			Type:   IngressControllerDeploymentRollingOutConditionType,
+			Status: operatorv1.ConditionTrue,
+			Reason: "DeploymentRollingOut",
+			Message: fmt.Sprintf(
+				"Waiting for router deployment rollout to finish: %d out of %d new replicas have been updated...\n",
+				deployment.Status.UpdatedReplicas, *deployment.Spec.Replicas),
+		}
+	}
+	// If have replicas greater than updated replicas, then we are waiting for old replicas to terminate.
+	if deployment.Status.Replicas > deployment.Status.UpdatedReplicas {
+		return operatorv1.OperatorCondition{
+			Type:   IngressControllerDeploymentRollingOutConditionType,
+			Status: operatorv1.ConditionTrue,
+			Reason: "DeploymentRollingOut",
+			Message: fmt.Sprintf(
+				"Waiting for router deployment rollout to finish: %d old replicas are pending termination...\n",
+				deployment.Status.Replicas-deployment.Status.UpdatedReplicas),
+		}
+	}
+	// If available replicas less than updated replicas, then we are waiting for updated replicas to become available.
+	if deployment.Status.AvailableReplicas < deployment.Status.UpdatedReplicas {
```

Comment:
It's fine to return only one condition (instead of appending conditions to a list), but maybe this one should be the second one evaluated, and the "pending termination" one be evaluated last?

Reply:
Actually, the more I think about this....shouldn't this be considered a Degraded condition, not  a DeploymentRollingOut condition?

Reply:
@candita Just wanted to recap the individual conditions types that we use for ingress.
This is the "Type: <type>" in the condition message of the IC object. **BOLDED** are status used for cluster operator status `oc get co` and in parentheses is the condition referenced in the calculation:

1. **Available**
    a. DeploymentAvailable (# 7)
    b. DNSReady (# 14)
    c. LoadBalancerReady (# 12)
2. **Progressing**
    a. LoadBalancerProgressing (# 16, NEW: Previously wasn't a separate condition)
    b. DeploymentRollingOut (# 15, NEW: Previously wasn't a separate condition)
3. **Degraded**
    a. PodsScheduled (# 6)
    b. DeploymentAvailable (# 7)
    c. DeploymentReplicasMinAvailable (# 8)
    d. DeploymentReplicasAllAvailable (# 9)
    e. LoadBalancerReady (# 12)
    f. DNSReady (# 14)
4. **Upgradeable**
    a. _Default Cert Check_ (NOT A SEPARATE CONDITION)
    b. _Load Balancer Upgradeable_ (NOT A SEPARATE CONDITION)
5. Admitted
6. PodsScheduled
7. DeploymentAvailable
8. DeploymentReplicasMinAvailable
9. DeploymentReplicasAllAvailable
10. CanaryChecksSucceeding
11. LoadBalancerManaged
12. LoadBalancerReady
13. DNSManaged
14. DNSReady
15. (NEW) DeploymentRollingOut
16. (NEW) LoadBalancerProgressing

Previously, 15 and 16 were stuffed into 3 (`Degraded`) and not individualized conditions. By this, I mean the conditions were not separate and were concatenated together into the `Degraded` condition's Message field, just like `Upgradeable` still is today. Does that help visualize the structure? 

To your point:

> Actually, the more I think about this....shouldn't this be considered a Degraded condition, not a DeploymentRollingOut condition?

"Rolliing Out" says `Progressing` to me. The `Degraded` condition makes sense the way it is calculated today in which it uses `DeploymentAvailable` and the other conditions. Do you disagree with that? If it helps, we can have a call to review ingress status conditions in their entirety

Reply:
Good overview.  This is worth posting to our FAQ.

Reply:
In fact, I would go so far as to recommend that we publish some kind of release note that describes the change to status.  I suppose there could be CI, monitoring, or other operators that rely on these statuses being unchanging over time, so we at least need to document the changes.

Reply:
More documentation is good.  I'm wary of documenting these in a way that commits us to maintaining backwards compatibility, but a release note and a FAQ would be useful,.  Maybe something along the lines of this to emphasize that these are not formally part of the API: 'To simplify the computation of the "Progressing" status condition, this fix introduces two new status conditions for internal use: "DeploymentRollingOut" and "LoadBalancerProgressing".'